src/stonkgs/api/example.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+
+"""Full example of how to use a fine tuned model.
+
+Run with: ``python -m stonkgs.api.example``
+"""
+
+import pickle
+import time
+
+import click
+import pandas as pd
+import pystow
+import torch
+from datasets import Dataset
+from tqdm import tqdm
+from transformers.trainer_utils import PredictionOutput
+
+from stonkgs import STonKGsForSequenceClassification, preprocess_df_for_embeddings
+from stonkgs.api.constants import ensure_embeddings, ensure_species, ensure_walks
+
+
+def _onrows(source_df: pd.DataFrame, model):
+    click.echo("Processing df for embeddings")
+    t = time.time()
+    preprocessed_df = preprocess_df_for_embeddings(
+        df=source_df,
+        embedding_name_to_vector_path=ensure_embeddings(),
+        embedding_name_to_random_walk_path=ensure_walks(),
+    )[["input_ids", "attention_mask", "token_type_ids"]]
+    click.echo(f"done processing df for embeddings after {time.time() - t:.2f} seconds")
+
+    dataset = Dataset.from_pandas(preprocessed_df)
+    dataset.set_format("torch")
+
+    # Initialize Softmax to process the logits predictions later on
+    softmax = torch.nn.Softmax(dim=1)
```

Comment:
@helena-balabin what's the advantage of using the modular form versus just applying the functional form?

Reply:
There's no specific reason other than my (bad) habits, you're right, the functional form would have totally been sufficient in this case.